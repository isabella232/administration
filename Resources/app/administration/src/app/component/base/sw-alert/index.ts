import type { NotificationType } from 'src/app/mixin/notification.mixin';
import type { PropType } from 'vue';
import template from './sw-alert.html.twig';
import './sw-alert.scss';

const { Component } = Shopware;
type AppearanceType = 'default' | 'notification' | 'system';
type CssClassesObject = { [key: string]: boolean };
type CssClasses = Array<string | CssClassesObject> | CssClassesObject;

/**
 * @description
 * The <u>sw-alert</u> component is used to convey important information to the user. It comes in 4 variations,
 * <strong>success</strong>, <strong>info</strong>, <strong>warning</strong> and <strong>error</strong>. These have
 * default icons assigned which can be changed and represent different actions
 * @status ready
 * @example-type dynamic
 * @component-example
 * <sw-alert variant="info" title="Example title" :closable="true">
 *    Sample text
 * </sw-alert>
 */
Component.register('sw-alert', {
    template,

    props: {
        variant: {
            type: String as PropType<NotificationType>,
            required: false,
            default: 'info',
            validValues: ['info', 'warning', 'error', 'success'],
            validator(value: string): boolean {
                return ['info', 'warning', 'error', 'success'].includes(value);
            },
        },
        appearance: {
            type: String as PropType<AppearanceType>,
            required: false,
            default: 'default',
            validValues: ['default', 'notification', 'system'],
            validator(value) {
                return ['default', 'notification', 'system'].includes(value);
            },
        },
        title: {
            type: String,
            required: false,
            default: '',
        },
        showIcon: {
            type: Boolean,
            required: false,
            // TODO: Boolean props should only be opt in and therefore default to false
            // eslint-disable-next-line vue/no-boolean-default
            default: true,
        },
        closable: {
            type: Boolean,
            required: false,
            default: false,
        },
        notificationIndex: {
            type: String,
            required: false,
            default: null,
        },
    },
    computed: {
        alertIcon(): string {
            const iconConfig: { [type: string]: string } = {
                info: 'default-badge-info',
                warning: 'default-badge-warning',
                error: 'default-badge-error',
                success: 'default-basic-checkmark-circle',
            };

            return iconConfig[this.variant] || 'default-bell-bell';
        },

        hasActionSlot(): boolean {
            return !!this.$slots.actions;
        },

        alertClasses(): CssClasses {
            return [
                `sw-alert--${this.variant}`,
                `sw-alert--${this.appearance}`,
                {
                    'sw-alert--icon': this.showIcon,
                    'sw-alert--no-icon': !this.showIcon,
                    'sw-alert--closable': this.closable,
                    'sw-alert--actions': this.hasActionSlot,
                },
            ];
        },

        alertBodyClasses(): CssClasses {
            return {
                'sw-alert__body--icon': this.showIcon,
                'sw-alert__body--closable': this.closable,
            };
        },
    },
});
