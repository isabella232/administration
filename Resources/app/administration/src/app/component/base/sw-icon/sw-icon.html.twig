{% block sw_icon %}
<component
    v-if="!replacementName || !useIconKit"
    :is="iconName"
    class="sw-icon"
    :class="classes"
    :style="styles"
    :aria-hidden="decorative"
    v-on="$listeners"
/>

<component
    v-else
    :is="replacementName"
    class="sw-icon"
    :class="classes"
    :style="styles"
    :aria-hidden="decorative"
    v-on="$listeners"
/>
{% endblock %}
