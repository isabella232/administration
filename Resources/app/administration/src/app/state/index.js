import notification from './notification.store';
import session from './session.store';
import system from './system.store';
import adminMenu from './admin-menu.store';
import context from './context.store';
import error from './error.store';
import settingsItems from './settings-item.store';
import shopwareApps from './shopware-apps.store';

export default {
    notification,
    session,
    system,
    adminMenu,
    context,
    error,
    settingsItems,
    shopwareApps
};
