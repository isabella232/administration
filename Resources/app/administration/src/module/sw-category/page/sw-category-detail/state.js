const { Criteria } = Shopware.Data;

export default {
    namespaced: true,

    state() {
        return {
            landingPage: null,
            category: null,
            customFieldSets: [],
            landingPagesToDelete: undefined,
            categoriesToDelete: undefined
        };
    },

    mutations: {
        setActiveLandingPage(state, { landingPage }) {
            state.landingPage = landingPage;
        },

        setActiveCategory(state, { category }) {
            state.category = category;
        },

        setCustomFieldSets(state, newCustomFieldSets) {
            state.customFieldSets = newCustomFieldSets;
        },

        setLandingPagesToDelete(state, { landingPagesToDelete }) {
            state.landingPagesToDelete = landingPagesToDelete;
        },

        setCategoriesToDelete(state, { categoriesToDelete }) {
            state.categoriesToDelete = categoriesToDelete;
        }
    },

    actions: {
        setActiveLandingPage({ commit }, payload) {
            commit('setActiveLandingPage', payload);
        },

        loadActiveLandingPage({ commit }, { repository, id, apiContext }) {
            if (id === 'create') {
                const landingPage = repository.create(apiContext);
                landingPage.cmsPageId = null;
                commit('setActiveLandingPage', { landingPage });
                return Promise.resolve();
            }

            const criteria = new Criteria();

            criteria.addAssociation('tags');
            criteria.addAssociation('salesChannels');

            return repository.get(id, apiContext, criteria).then((landingPage) => {
                commit('setActiveLandingPage', { landingPage });
            });
        },

        setActiveCategory({ commit }, payload) {
            commit('setActiveCategory', payload);
        },

        loadActiveCategory({ commit }, { repository, id, apiContext }) {
            const criteria = new Criteria();

            criteria.getAssociation('seoUrls')
                .addFilter(Criteria.equals('isCanonical', true));

            criteria.addAssociation('tags')
                .addAssociation('media')
                .addAssociation('navigationSalesChannels')
                .addAssociation('serviceSalesChannels')
                .addAssociation('footerSalesChannels');


            return repository.get(id, apiContext, criteria).then((category) => {
                commit('setActiveCategory', { category });
            });
        }
    }
};
