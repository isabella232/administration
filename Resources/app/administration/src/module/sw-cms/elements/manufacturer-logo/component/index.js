const { Component, Mixin, Utils } = Shopware;

Component.extend('sw-cms-el-manufacturer-logo', 'sw-cms-el-image', {
    mixins: [
        Mixin.getByName('cms-element')
    ],

    computed: {
        isProductPage() {
            return Utils.get(this.cmsPageState, 'currentPage.type', '') === 'product_detail';
        }
    },

    methods: {
        createdComponent() {
            this.initElementConfig('manufacturer-logo');
            this.initElementData('manufacturer-logo');

            if (this.isProductPage) {
                this.element.config.media.source = 'mapped';
                this.element.config.media.value = 'product.manufacturer.media';
            }
        }
    }
});
