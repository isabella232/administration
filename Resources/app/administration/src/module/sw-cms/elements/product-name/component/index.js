const { Component, Mixin, Utils } = Shopware;

Component.extend('sw-cms-el-product-name', 'sw-cms-el-text', {
    mixins: [
        Mixin.getByName('cms-element')
    ],

    computed: {
        isProductPage() {
            return Utils.get(this.cmsPageState, 'currentPage.type', '') === 'product_detail';
        }
    },

    methods: {
        createdComponent() {
            this.initElementConfig('product-name');

            if (this.isProductPage) {
                this.element.config.content.source = 'mapped';
                this.element.config.content.value = 'product.name';
            }
        }
    }
});
