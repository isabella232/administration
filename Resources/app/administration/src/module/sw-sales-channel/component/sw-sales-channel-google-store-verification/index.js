import template from './sw-sales-channel-google-store-verification.html.twig';
import { getErrorMessage } from '../../helper/get-error-message.helper';

import './sw-sales-channel-google-store-verification.scss';


const { Component, Service, Utils, Mixin, State } = Shopware;
const { mapState, mapGetters } = Component.getComponentHelper();

Component.register('sw-sales-channel-google-store-verification', {
    template,

    mixins: [
        Mixin.getByName('notification')
    ],

    props: {
        salesChannel: {
            type: Object,
            required: true
        }
    },

    data() {
        return {
            items: [
                {
                    status: 'info',
                    key: 'siteIsVerified',
                    errorLink: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.linkVerifyWebsite'),
                    description: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textVerifiedWebsite')
                },
                {
                    status: 'info',
                    key: 'shoppingAdsPolicies',
                    errorLink: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textAdsPolicyUrl'),
                    description: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textAdsPolicy')
                },
                {
                    status: 'info',
                    key: 'contactPage',
                    description: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textAccurateContact')
                },
                {
                    status: 'info',
                    key: 'secureCheckoutProcess',
                    description: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textSecureCheckoutProcess')
                },
                {
                    status: 'info',
                    key: 'revocationPage',
                    description: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textReturnPolicy')
                },
                {
                    status: 'info',
                    key: 'shippingPaymentInfoPage',
                    description: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textBillingTerms')
                },
                {
                    status: 'info',
                    key: 'completeCheckoutProcess',
                    description: this.$tc('sw-sales-channel.modalGooglePrograms.step-4.textComplete')
                }
            ],
            isLoading: false
        };
    },

    watch: {
        isIncompleteVerification: {
            handler: 'updateButtons'
        }
    },

    computed: {
        ...mapState('swSalesChannel', [
            'storeVerification'
        ]),

        ...mapGetters('swSalesChannel', [
            'isIncompleteVerification'
        ])
    },

    created() {
        this.createdComponent();
    },

    mounted() {
        this.mountedComponent();
    },

    methods: {
        createdComponent() {
            this.updateButtons();
        },

        mountedComponent() {
            if (!this.storeVerification) {
                this.verifyStore();
            } else {
                this.updateItems(this.storeVerification);
            }
        },

        updateButtons() {
            const buttonConfig = {
                right: {
                    label: this.$tc('sw-sales-channel.modalGooglePrograms.buttonNext'),
                    variant: 'primary',
                    action: 'sw.sales.channel.detail.base.step-5',
                    disabled: this.isIncompleteVerification
                },
                left: {
                    label: this.$tc('sw-sales-channel.modalGooglePrograms.buttonBack'),
                    variant: null,
                    action: 'sw.sales.channel.detail.base.step-3',
                    disabled: false
                }
            };

            this.$emit('buttons-update', buttonConfig);
        },

        async verifyStore() {
            this.isLoading = true;

            try {
                const { data: storeVerification } = await Service('googleShoppingService').verifyStore(this.salesChannel.id);
                const storeVerificationData = Utils.get(storeVerification, 'data', null);

                State.commit('swSalesChannel/setStoreVerification', storeVerificationData);

                this.updateItems(storeVerificationData);
            } catch (error) {
                this.showErrorNotification(error);
            } finally {
                this.isLoading = false;
            }
        },

        getIconName(status) {
            if (status === 'success') return 'small-default-checkmark-line-medium';
            if (status === 'danger') return 'small-default-x-line-medium';

            return 'small-default-circle-medium';
        },

        showErrorNotification(error) {
            const errorDetail = getErrorMessage(error);

            this.createNotificationError({
                title: this.$tc('global.default.error'),
                message: errorDetail || this.$tc('global.notification.unspecifiedSaveErrorMessage')
            });
        },

        getFixPath(item) {
            if (item.key === 'completeCheckoutProcess' || item.key === 'secureCheckoutProcess') {
                return {
                    name: 'sw.sales.channel.detail',
                    params: { id: this.salesChannel.productExports[0].storefrontSalesChannelId }
                };
            }

            return { name: 'sw.settings.basic.information.index' };
        },

        updateItems(storeVerification) {
            this.items = this.items.map((item) => {
                return {
                    ...item,
                    status: storeVerification[item.key] ? 'success' : 'danger'
                };
            });
        }
    }
});
