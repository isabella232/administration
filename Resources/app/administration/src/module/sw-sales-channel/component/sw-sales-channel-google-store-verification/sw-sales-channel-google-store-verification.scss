@import "~scss/variables";
@import "~scss/mixins";

.sw-sales-channel-google-store-verification {
    &__top {
        padding-top: 0;
    }

    &__check-list {
        padding: 10px 0;
    }

    .sw-sales-channel-google-store-verification__check-item {
        margin: 20px 0;
        display: flex;
        align-items: center;
        position: relative;

        .sw-label {
            margin: 0;
            flex-shrink: 0;
            display: flex;
            align-items: center;
            justify-content: center;
            text-align: center;

            @include circle(32px);
        }

        &-description {
            padding-left: 20px;
            font-size: $font-size-small;
            line-height: 32px;
        }
    }

    &__fix-link {
        text-decoration: none;
        color: $color-shopware-brand-500;
        font-size: $font-size-small;
        margin-left: 10px;

        &:hover {
            color: $color-shopware-brand-600;
            text-decoration: underline;
        }
    }

    &__bottom {
        padding-bottom: 0;
    }
}
