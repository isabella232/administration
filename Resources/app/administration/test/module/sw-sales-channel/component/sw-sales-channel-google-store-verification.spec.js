import { shallowMount } from '@vue/test-utils';
import 'src/module/sw-sales-channel/component/sw-sales-channel-google-store-verification';
import 'src/app/component/base/sw-label';
import 'src/app/component/base/sw-icon';
import 'src/app/component/base/sw-button';
import state from 'src/module/sw-sales-channel/state/salesChannel.store';

Shopware.State.registerModule('swSalesChannel', state);

describe('module/sw-sales-channel/component/sw-sales-channel-google-store-verification', () => {
    let wrapper;

    beforeAll(() => {
        const errorResponse = {
            response: {
                data: {
                    errors: [
                        {
                            code: 'This is an error code',
                            detail: 'This is an detailed error message'
                        }
                    ]
                }
            }
        };

        Shopware.Service().register('googleShoppingService', () => {
            const response = {
                data: {
                    data: {
                        siteIsVerified: true,
                        shoppingAdsPolicies: true,
                        contactPage: false,
                        secureCheckoutProcess: true,
                        revocationPage: true,
                        shippingPaymentInfoPage: true,
                        completeCheckoutProcess: false
                    }
                }
            };

            return {
                verifyStore: (salesChannelId) => {
                    if (salesChannelId) {
                        return Promise.resolve(response);
                    }

                    // eslint-disable-next-line prefer-promise-reject-errors
                    return Promise.reject(errorResponse);
                }
            };
        });
    });

    beforeEach(() => {
        wrapper = shallowMount(Shopware.Component.build('sw-sales-channel-google-store-verification'), {
            store: Shopware.State._store,
            stubs: {
                'sw-label': Shopware.Component.build('sw-label'),
                'sw-icon': true,
                'sw-sales-channel-detail-protect-link': true,
                'sw-button': true,
                'router-link': true
            },
            mocks: {
                $tc: (translationPath) => translationPath,
                $route: { name: 'sw.sales.channel.detail.base.step-4' },
                $router: { push: () => {} }
            },
            propsData: {
                salesChannel: {
                    id: 1,
                    productExports: [{
                        storefrontSalesChannelId: 2
                    }]
                }
            }
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it('should render number of item which equals total store verification items length', async () => {
        wrapper.setData({
            items: [
                {
                    status: 'success',
                    description: 'Shopping Ads policy'
                },
                {
                    status: 'success',
                    description: 'Accurate contact information'
                },
                {
                    status: 'success',
                    description: 'Complete checkout process'
                },
                {
                    status: 'danger',
                    description: 'Return policy'
                }
            ]
        });

        await wrapper.vm.$nextTick();

        const verifiedItems = wrapper.findAll('.sw-sales-channel-google-store-verification__check-item');
        expect(verifiedItems.length).toEqual(wrapper.vm.items.length);
    });

    it('should render item status and description correctly', async () => {
        wrapper.setData({
            items: [
                {
                    status: 'success',
                    key: 'siteIsVerified',
                    description: 'Verified storefront domain'
                },
                {
                    status: 'danger',
                    key: 'shoppingAdsPolicies',
                    description: 'Shopping Ads policy'
                },
                {
                    status: 'danger',
                    key: 'contactPage',
                    description: 'Accurate contact information'
                },
                {
                    status: 'success',
                    key: 'secureCheckoutProcess',
                    description: 'Complete checkout process'
                },
                {
                    status: 'success',
                    key: 'revocationPage',
                    description: 'Return policy'
                },
                {
                    status: 'success',
                    key: 'shippingPaymentInfoPage',
                    description: 'Billing terms and conditions'
                }
            ]
        });

        await wrapper.vm.$nextTick();

        expect(wrapper.element).toMatchSnapshot();
    });

    it('items should update correctly when verifyStore is successful', async () => {
        wrapper.setMethods({ createNotificationError: jest.fn() });
        wrapper.setData({
            items: [
                {
                    status: 'info',
                    key: 'siteIsVerified',
                    description: 'Verified storefront domain'
                },
                {
                    status: 'info',
                    key: 'shoppingAdsPolicies',
                    description: 'Shopping Ads policy'
                },
                {
                    status: 'info',
                    key: 'contactPage',
                    description: 'Accurate contact information'
                },
                {
                    status: 'info',
                    key: 'secureCheckoutProcess',
                    description: 'Complete checkout process'
                },
                {
                    status: 'info',
                    key: 'revocationPage',
                    description: 'Return policy'
                },
                {
                    status: 'info',
                    key: 'shippingPaymentInfoPage',
                    description: 'Billing terms and conditions'
                },
                {
                    status: 'info',
                    key: 'completeCheckoutProcess',
                    description: 'Complete checkout process'
                }
            ]
        });

        await wrapper.vm.$nextTick();
        await wrapper.vm.verifyStore();

        const newItems = [
            {
                status: 'success',
                key: 'siteIsVerified',
                description: 'Verified storefront domain'
            },
            {
                status: 'success',
                key: 'shoppingAdsPolicies',
                description: 'Shopping Ads policy'
            },
            {
                status: 'danger',
                key: 'contactPage',
                description: 'Accurate contact information'
            },
            {
                status: 'success',
                key: 'secureCheckoutProcess',
                description: 'Complete checkout process'
            },
            {
                status: 'success',
                key: 'revocationPage',
                description: 'Return policy'
            },
            {
                status: 'success',
                key: 'shippingPaymentInfoPage',
                description: 'Billing terms and conditions'
            },
            {
                status: 'danger',
                key: 'completeCheckoutProcess',
                description: 'Complete checkout process'
            }
        ];

        expect(wrapper.vm.items).toStrictEqual(newItems);
    });

    it('showErrorNotification should be called update when verifyStore is failed', async () => {
        wrapper.setProps({
            salesChannel: {
                ...wrapper.vm.salesChannel,
                id: ''
            }
        });
        wrapper.setMethods({ createNotificationError: jest.fn() });

        const spy = jest.spyOn(wrapper.vm, 'showErrorNotification');

        await wrapper.vm.$nextTick();
        await wrapper.vm.verifyStore();

        expect(spy).toHaveBeenCalled();
    });
});
