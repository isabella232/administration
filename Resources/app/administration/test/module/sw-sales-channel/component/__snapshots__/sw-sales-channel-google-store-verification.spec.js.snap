// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module/sw-sales-channel/component/sw-sales-channel-google-store-verification should render item status and description correctly 1`] = `
<div
  class="sw-sales-channel-google-store-verification"
>
  <div
    class="sw-sales-channel-google-store-verification__top"
  >
    <strong>
      sw-sales-channel.modalGooglePrograms.step-4.titleGeneral
    </strong>
     
    <div
      class="sw-sales-channel-google-store-verification__check-list"
    >
      <div
        class="sw-sales-channel-google-store-verification__check-item"
      >
        <span
          class="sw-label sw-label--appearance-circle sw-label--size-default sw-label--success"
        >
          <span
            class="sw-label__caption"
          >
            <sw-icon-stub
              name="small-default-checkmark-line-medium"
              size="16px"
            />
          </span>
           
          <!---->
        </span>
         
        <span
          class="sw-sales-channel-google-store-verification__check-item-description"
        >
          
                                Verified storefront domain

                                
          <!---->
        </span>
      </div>
      <div
        class="sw-sales-channel-google-store-verification__check-item"
      >
        <span
          class="sw-label sw-label--appearance-circle sw-label--size-default sw-label--danger"
        >
          <span
            class="sw-label__caption"
          >
            <sw-icon-stub
              name="small-default-x-line-medium"
              size="16px"
            />
          </span>
           
          <!---->
        </span>
         
        <span
          class="sw-sales-channel-google-store-verification__check-item-description"
        >
          
                                Shopping Ads policy

                                
          <sw-sales-channel-detail-protect-link-stub
            class="sw-sales-channel-google-store-verification__fix-link"
          >
            
                                        sw-sales-channel.modalGooglePrograms.step-4.textFixHere
                                    
          </sw-sales-channel-detail-protect-link-stub>
        </span>
      </div>
      <div
        class="sw-sales-channel-google-store-verification__check-item"
      >
        <span
          class="sw-label sw-label--appearance-circle sw-label--size-default sw-label--danger"
        >
          <span
            class="sw-label__caption"
          >
            <sw-icon-stub
              name="small-default-x-line-medium"
              size="16px"
            />
          </span>
           
          <!---->
        </span>
         
        <span
          class="sw-sales-channel-google-store-verification__check-item-description"
        >
          
                                Accurate contact information

                                
          <router-link-stub
            class="sw-sales-channel-google-store-verification__fix-link"
            rel="noopener"
            target="_blank"
            to="[object Object]"
          >
            
                                        sw-sales-channel.modalGooglePrograms.step-4.textFixHere
                                    
          </router-link-stub>
        </span>
      </div>
      <div
        class="sw-sales-channel-google-store-verification__check-item"
      >
        <span
          class="sw-label sw-label--appearance-circle sw-label--size-default sw-label--success"
        >
          <span
            class="sw-label__caption"
          >
            <sw-icon-stub
              name="small-default-checkmark-line-medium"
              size="16px"
            />
          </span>
           
          <!---->
        </span>
         
        <span
          class="sw-sales-channel-google-store-verification__check-item-description"
        >
          
                                Complete checkout process

                                
          <!---->
        </span>
      </div>
      <div
        class="sw-sales-channel-google-store-verification__check-item"
      >
        <span
          class="sw-label sw-label--appearance-circle sw-label--size-default sw-label--success"
        >
          <span
            class="sw-label__caption"
          >
            <sw-icon-stub
              name="small-default-checkmark-line-medium"
              size="16px"
            />
          </span>
           
          <!---->
        </span>
         
        <span
          class="sw-sales-channel-google-store-verification__check-item-description"
        >
          
                                Return policy

                                
          <!---->
        </span>
      </div>
      <div
        class="sw-sales-channel-google-store-verification__check-item"
      >
        <span
          class="sw-label sw-label--appearance-circle sw-label--size-default sw-label--success"
        >
          <span
            class="sw-label__caption"
          >
            <sw-icon-stub
              name="small-default-checkmark-line-medium"
              size="16px"
            />
          </span>
           
          <!---->
        </span>
         
        <span
          class="sw-sales-channel-google-store-verification__check-item-description"
        >
          
                                Billing terms and conditions

                                
          <!---->
        </span>
      </div>
    </div>
     
    <sw-button-stub
      size="small"
      variant="ghost"
    >
      
                        sw-sales-channel.modalGooglePrograms.step-4.buttonCheckAgain
                    
    </sw-button-stub>
  </div>
</div>
`;
